tests/terraform/testutils.go

```diff
@@ -0,0 +1,272 @@
+	out, err := c.CombinedOutput()
+	return string(out), err
+}
+
+// Used to count the pods using prefix passed in the list of pods
+func CountOfStringInSlice(str string, pods []Pod) int {
+	count := 0
+	for _, pod := range pods {
+		if strings.Contains(pod.Name, str) {
+			count++
+		}
+	}
+	return count
+}
+
+func DeployWorkload(workload, kubeconfig string) (string, error) {
+	resourceDir := GetBasepath() + "/tests/terraform/resource_files"
+	files, err := ioutil.ReadDir(resourceDir)
+	if err != nil {
+		err = fmt.Errorf("%s : Unable to read resource manifest file for %s", err, workload)
+		return "", err
+	}
+	for _, f := range files {
+		filename := filepath.Join(resourceDir, f.Name())
+		if strings.TrimSpace(f.Name()) == workload {
+			cmd := "kubectl apply -f " + filename + " --kubeconfig=" + kubeconfig
+			return RunCommand(cmd)
+		}
+	}
+	return "", nil
+}
+
+func RemoveWorkload(workload, kubeconfig string) (string, error) {
+	resourceDir := GetBasepath() + "/tests/terraform/resource_files"
+	files, err := ioutil.ReadDir(resourceDir)
+	if err != nil {
+		err = fmt.Errorf("%s : Unable to read resource manifest file for %s", err, workload)
+		return "", err
+	}
+	for _, f := range files {
+		filename := filepath.Join(resourceDir, f.Name())
+		if strings.TrimSpace(f.Name()) == workload {
+			cmd := "kubectl delete -f " + filename + " --kubeconfig=" + kubeconfig
+			return RunCommand(cmd)
+		}
+	}
+	return "", nil
+}
+
+func FetchClusterIP(kubeconfig string, namespace string, servicename string) (string, string, error) {
+	ipCmd := "kubectl get svc " + servicename + " -n " + namespace + " -o jsonpath='{.spec.clusterIP}' --kubeconfig=" + kubeconfig
+	ip, err := RunCommand(ipCmd)
+	if err != nil {
+		return "", "", err
+	}
+	portCmd := "kubectl get svc " + servicename + " -n " + namespace + " -o jsonpath='{.spec.ports[0].port}' --kubeconfig=" + kubeconfig
+	port, err := RunCommand(portCmd)
+	if err != nil {
+		return "", "", err
+	}
+	return ip, port, err
+}
+
+func FetchNodeExternalIP(kubeconfig string) []string {
+	cmd := "kubectl get node --output=jsonpath='{range .items[*]} { .status.addresses[?(@.type==\"ExternalIP\")].address}' --kubeconfig=" + kubeconfig
+	time.Sleep(10 * time.Second)
+	res, _ := RunCommand(cmd)
+	nodeExternalIP := strings.Trim(res, " ")
+	nodeExternalIPs := strings.Split(nodeExternalIP, " ")
+	return nodeExternalIPs
+}
+
+func FetchIngressIP(namespace string, kubeconfig string) ([]string, error) {
+	cmd := "kubectl get ingress -n " + namespace + " -o jsonpath='{.items[0].status.loadBalancer.ingress[*].ip}' --kubeconfig=" + kubeconfig
+	res, err := RunCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	ingressIP := strings.Trim(res, " ")
+	if ingressIP != "" {
+		ingressIPs := strings.Split(ingressIP, " ")
+		return ingressIPs, nil
+	} else {
+		return nil, nil
+	}
+}
+
+func parseNodes(kubeConfig string, print bool, cmd string) ([]Node, error) {
+	nodes := make([]Node, 0, 10)
+	nodeList := ""
```

Comment:
Since Go is statically and strongly typed, it's an antipattern to use Hungarian notation.